plugin/completion/flags_manager.py

```diff
@@ -346,44 +380,47 @@ def flags_from_clang_file(file, separate_includes):
         flags = set()
         with open(file.full_path()) as f:
             content = f.readlines()
-            flags = FlagsManager.parse_flags(file.folder(),
-                                             content,
-                                             separate_includes)
+            flags = FlagsManager.parse_flags(file.folder(), content)
         log.debug(" .clang_complete contains flags: %s", flags)
         return flags
 
     @staticmethod
-    def parse_flags(folder, lines, separate_includes):
+    def parse_flags(folder, lines):
         """
         Parse the flags in a given file
 
         Args:
             folder (str): current folder
             lines (str[]): lines to parse
-            separate_includes (bool): if True "-I/blah" turns to "-I '/blah'"
 
         Returns:
             str[]: flags
         """
-        mask = '{}{}'
-        separate_mask = '{} "{}"'
+
+        def split_if_include(flag):
+            """ Split flag if it is an include flag
+
+            Args:
+                flag (str): flag to test and split if needed
+
+            Returns:
+                (str, str): tuple of (prefix, path)
+            """
+            for prefix in FlagsManager._include_prefixes:
+                if flag.startswith(prefix):
+                    return (prefix, flag[len(prefix):])
```

Comment:
Currently for include that are already absolute the code does:
a) split string into two parts
b) do nothing if path is already absolute
c) concatenate strings back together

With doing the check inside the method you can optimize it to just be:
a) do nothing if path is already absolute using `path.isabs(flag[len(prefix)])`